cdap-master/src/main/java/co/cask/cdap/data/tools/UpgradeTool.java

```diff
@@ -505,4 +594,52 @@ private FileMetaDataManager getFileMetaDataManager() {
     }
     return fileMetaDataManager;
   }
+
+  /**
+   * gets the {@link AdapterService}
+   *
+   * @return {@link AdapterService}
+   */
+  private AdapterService getAdapterService() {
+    if (adapterService == null) {
+      adapterService = injector.getInstance(AdapterService.class);
+    }
+    return adapterService;
+  }
+
+  /**
+   * gets the {@link DatasetBasedTimeScheduleStore} which stores the schedules of adapters
+   *
+   * @return {@link DatasetBasedTimeScheduleStore}
+   * @throws SchedulerException
+   */
+  private DatasetBasedTimeScheduleStore getDatasetBasedTimeScheduleStore() throws SchedulerException {
+    if (datasetBasedTimeScheduleStore == null) {
+      datasetBasedTimeScheduleStore = injector.getInstance(Key.get(DatasetBasedTimeScheduleStore.class,
+                                                                   Names.named("datasetBasedTimeScheduleStore")));
+      // need to call initialize on datasetBasedTimeScheduleStore
+      ExecutorThreadPool threadPool = new ExecutorThreadPool(1);
```

Comment:
This is to create and initialize Quartz scheduler which is needed to initialize the DatasetBasedTimeScheduleStore. 
This is present AppFabricServiceRuntimeModule#getScheduler which we use in injection but is not exposed and is used to get the Supplier. I talked to @albertshau and we were not able to find out a way to get/reuse that one.
@chtyim : Any ideas ? Thanks.